solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -2424,6 +2430,32 @@ public PlacementPluginFactory<? extends PlacementPluginConfig> getPlacementPlugi
   public void runAsync(Runnable r) {
     coreContainerAsyncTaskExecutor.submit(r);
   }
+
+  public static void setWeakStringInterner() {
+    Interner<String> interner = Interner.newWeakInterner();
+    ClusterState.setStrInternerParser(
+        new Function<>() {
```

Comment:
Let's move this function to `Utils` with the other ObjectBuilders.

Reply:
utils is in the solrj and solrj does not have visibility to caffeine classes